kratos/includes/initial_state.h

```diff
@@ -0,0 +1,279 @@
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class InitialState
+ * @ingroup StructuralMechanicsApplication
+ * @brief Define the initial state of the material in terms of initial stress/strain/F
+ * @details Storages the information regarding initial stresses/strains/F
+ * @author Alejandro Cornejo
+ */
+class KRATOS_API(KRATOS_CORE) InitialState
+{
+    public:
+        ///@name Type Definitions
+        ///@{
+        typedef std::size_t SizeType;
+
+        enum class InitialImposingType {StrainOnly = 0, StressOnly = 1, DeformationGradientOnly = 2, StrainAndStress = 3, DeformationGradientAndStress = 4};
+
+        /// Pointer definition of NodeSearchUtility
+        KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(InitialState);
+
+        ///@}
+        ///@name Life Cycle
+        ///@{
+
+        /// Default constructor.
+        InitialState()
+        {}
+
+        /// Only defining Dimension constructor.
+        InitialState(const SizeType Dimension)
+        {
+            const SizeType voigt_size = (Dimension == 3) ? 6 : 3;
+            mInitialStressVector.resize(voigt_size, false);
+            mInitialStrainVector.resize(voigt_size, false);
+            mInitialDeformationGradientMatrix.resize(Dimension, Dimension, false);
+
+            noalias(mInitialStressVector) = ZeroVector(voigt_size);
+            noalias(mInitialStrainVector) = ZeroVector(voigt_size);
+            noalias(mInitialDeformationGradientMatrix) = ZeroMatrix(Dimension, Dimension);
+        }
+
+        // Full constructor
+        InitialState(const Vector& rInitialStrainVector,
+                     const Vector& rInitialStressVector,
+                     const Matrix& rInitialDeformationGradientMatrix)
+        {
+            const SizeType voigt_size = rInitialStrainVector.size();
+            const SizeType dimension  = rInitialDeformationGradientMatrix.size1();
+
+            KRATOS_ERROR_IF(voigt_size <= 0 || dimension <= 0) << "The imposed vector or matrix is null..." << std::endl;
+
+            mInitialStressVector.resize(voigt_size, false);
+            mInitialStrainVector.resize(voigt_size, false);
+            mInitialDeformationGradientMatrix.resize(dimension, dimension, false);
+
+            noalias(mInitialStressVector) = rInitialStressVector;
+            noalias(mInitialStrainVector) = rInitialStrainVector;
+            noalias(mInitialDeformationGradientMatrix) = rInitialDeformationGradientMatrix;
+        }
+
+        // Selective constructor for vectors
+        InitialState(const Vector& rImposingEntity,
+                     const int InitialImposingType = static_cast<int>(InitialImposingType::StrainOnly))
+        {
+            const SizeType voigt_size = rImposingEntity.size();
+            const SizeType dimension = (voigt_size == 6) ? 3 : 2;
+
+            mInitialStrainVector.resize(voigt_size, false);
+            mInitialStressVector.resize(voigt_size, false);
```

Comment:
I would argue that in this case you would best not resize what you don't need...